django/http/request.py

```diff
@@ -78,6 +87,20 @@ def __repr__(self):
             self.get_full_path(),
         )
 
+    def __getstate__(self):
+        obj_dict = self.__dict__.copy()
+        for attr in self.non_picklable_attrs:
+            if attr in obj_dict:
+                del obj_dict[attr]
+        return obj_dict
+
+    def __deepcopy__(self, memo):
+        obj = copy.copy(self)
+        obj.environ = copy.deepcopy(self.environ, memo)
+        obj.META = copy.deepcopy(self.META, memo)
```

Comment:
We should have a more generic solution and deep-copy all non-picklable attributes, e.g.
```python
        for attr in self.non_picklable_attrs:
            if hasattr(self, attr):
                setattr(obj, attr, copy.deepcopy(getattr(self, attr), memo))
```